### PR TITLE
Add change secretary general transaction and refactor three new proposal transaction codes

### PR DESCRIPTION
**Pull request by Houshoupei84**
Refactor three transaction codes: Secretary-General, Close Proposal, and Change Proposal Owner

### CODE REVIEW

File: blockchain/txvalidator.go [hunk 1]
```diff
@@ -1817,6 +1819,17 @@ func getCode(publicKey []byte) ([]byte, error) {
 			return redeemScript, nil
 		}
 	}
+	//<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (RainFallsSilent) [reply blockchain/txvalidator.go]: He has solved this problem in the second commit.

File: blockchain/txvalidator.go [hunk 2]
```diff
@@ -1817,6 +1819,17 @@ func getCode(publicKey []byte) ([]byte, error) {
...
+	//=======
+}
+
+func getDiDFromPublicKey(publicKey []byte) (*common.Uint168, error) {
+	if code, err := getCode(publicKey); err != nil {
+		return nil, err
+	} else {
+		return getDIDFromCode(code)
+	}
+	//>>>>>>> Add change secretary general proposal transaction
```
[the remark this answers is not in the document]
Comment (RainFallsSilent) [reply blockchain/txvalidator.go]: He has solved this problem in the second commit.

File: cmd/script/api/payloadtype.go [hunk 3]
```diff
@@ -965,6 +967,115 @@ func RegisterCRCCloseProposalHashType(L *lua.LState) {
...
+	if err != nil {
+		fmt.Println("wrong draft proposal hash")
+		os.Exit(1)
+	}
+
+	account := client.GetMainAccount()
+	CRCouncilMembercode := account.RedeemScript
+	CRCouncilMemberDID, _ := getDIDFromCode(CRCouncilMembercode)
+
+	fmt.Printf("account %+v\n", account)
```
Comment (peterwillcn) [inline cmd/script/api/payloadtype.go:1011]: Remove debug fmt.Printf

Comment (RainFallsSilent) [reply cmd/script/api/payloadtype.go:1011]: This is meant for testing and I don't think they need to be modified

Comment (RainFallsSilent) [reply cmd/script/api/payloadtype.go:1011]: This is meant for testing, and I don't think they need to be modified here

**Pull request merged**